sale_order_import/wizard/sale_order_import.py

```diff
@@ -0,0 +1,667 @@
+# Copyright 2016-2017 Akretion
+# @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# Copyright 2022 Camptocamp
+# @author: Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+import mimetypes
+from base64 import b64decode, b64encode
+
+from lxml import etree
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.osv.expression import AND
+from odoo.tools import config, float_compare, float_is_zero
+
+logger = logging.getLogger(__name__)
+
+
+class SaleOrderImport(models.TransientModel):
+    _name = "sale.order.import"
+    _description = "Sale Order Import from Files"
+
+    state = fields.Selection(
+        [("import", "Import"), ("update", "Update")], default="import"
+    )
+    partner_id = fields.Many2one("res.partner", string="Customer")
+    import_type = fields.Selection(
+        [("xml", "XML"), ("pdf", "PDF")],
+        required=True,
+        default=None,
+        help="Select a type which you want to import",
+    )
+    order_file = fields.Binary(
+        string="Request for Quotation or Order",
+        required=True,
+        help="Upload a Request for Quotation or an Order file. Supported "
+        "formats: XML and PDF (PDF with an embeded XML file).",
+    )
+    order_filename = fields.Char(string="Filename")
+    doc_type = fields.Selection(
+        [("rfq", "Request For Quotation"), ("order", "Sale Order")],
+        string="Document Type",
+        readonly=True,
+    )
+    price_source = fields.Selection(
+        [("pricelist", "Pricelist"), ("order", "Customer Order")],
+        string="Apply Prices From",
+    )
+    # for state = update
+    commercial_partner_id = fields.Many2one(
+        "res.partner", string="Commercial Entity", readonly=True
+    )
+    partner_shipping_id = fields.Many2one(
+        "res.partner", string="Shipping Address", readonly=True
+    )
+    sale_id = fields.Many2one("sale.order", string="Quotation to Update")
+
+    @api.onchange("order_file")
+    def order_file_change(self):
+        if not self.order_filename or not self.order_file:
+            self.doc_type = False
+            return
+
+        doc_type = self._parse_file(
+            self.order_filename, b64decode(self.order_file), detect_doc_type=True
+        )
+        if doc_type is None:
+            return {"warning": self._unsupported_file_msg(self.order_filename)}
+        self.doc_type = doc_type
+
+    def _parse_file(self, filename, filecontent, detect_doc_type=False):
+        assert filename, "Missing filename"
+        assert filecontent, "Missing file content"
+        filetype = mimetypes.guess_type(filename)
+        logger.debug("Order file mimetype: %s", filetype)
+        mimetype = filetype[0]
+        supported_types = {
```

Comment:
@QuocDuong1306 there should be a dedicated method to get types, peple can inherit it then